app/controllers/api/authentications_controller.rb

```diff
@@ -33,5 +41,16 @@ def build_additional_fields
         :credential_types => ::Authentication.build_credential_options
       }
     end
+
+    def validate_auth_attrs(data)
+      raise 'must supply a manager resource' unless data['manager_resource']
+      attrs = data.dup.except('manager_resource')
+      manager_resource = if data['manager_resource'].key?('href')
+                           parse_href(data['manager_resource']['href']).last
+                         elsif data['manager_resource'].key?('id')
+                           data['manager_resource']['id']
+                         end
+      [manager_resource, attrs]
```

Comment:
lines 48-52 could be replaced with:

```
manager_collection, manager_id = parse_href(data['manager_resource']['href']
raise 'invalid manger_resource href specified' unless manager_collection && manager_id
manager_resource = resource_search(manager_id, manager_collection, collection_class(manager_collection)
```

Not sure we want to support manager_resource id references since it would be vague for handling multiple types.